src/io/flutter/run/daemon/DevToolsService.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright 2020 The Chromium Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style license that can be
+ * found in the LICENSE file.
+ */
+package io.flutter.run.daemon;
+
+import com.google.common.collect.ImmutableList;
+import com.intellij.execution.ExecutionException;
+import com.intellij.execution.configurations.GeneralCommandLine;
+import com.intellij.execution.process.ProcessHandler;
+import com.intellij.openapi.components.ServiceManager;
+import com.intellij.openapi.project.Project;
+import com.intellij.openapi.project.ProjectManager;
+import com.intellij.openapi.project.ProjectManagerListener;
+import com.intellij.openapi.util.io.FileUtil;
+import com.intellij.openapi.vfs.CharsetToolkit;
+import io.flutter.FlutterUtils;
+import io.flutter.android.IntelliJAndroidSdk;
+import io.flutter.bazel.Workspace;
+import io.flutter.bazel.WorkspaceCache;
+import io.flutter.sdk.FlutterSdk;
+import io.flutter.sdk.FlutterSdkUtil;
+import io.flutter.utils.MostlySilentOsProcessHandler;
+import org.jetbrains.annotations.NotNull;
+
+import static com.jetbrains.lang.dart.ide.runner.server.vmService.VmServiceWrapper.LOG;
+
+public class DevToolsService {
+  private class DevToolsServiceListener implements DaemonEvent.Listener {
+  }
+
+  @NotNull private final Project project;
+  private DaemonApi daemonApi;
+  private ProcessHandler process;
+  private DevToolsInstance devToolsInstance;
```

Comment:
How can a caller of this know whether the devtools instance has started yet or not?
Perhaps vend a CompletableFuture with the devToolsInstance or a boolean indicating the service has started.
Also, how is the service stopped iff a project is closed?